pkg/operator/ceph/object/user/controller.go

```diff
@@ -294,14 +294,23 @@ func (r *ReconcileObjectStoreUser) createorUpdateCephUser(u *cephv1.CephObjectSt
 }
 
 func (r *ReconcileObjectStoreUser) isObjectStoreInitialized(u *cephv1.CephObjectStoreUser) error {
-	objContext := object.NewContext(r.context, r.clusterInfo, u.Spec.Store)
-	r.objContext = objContext
-
 	err := r.objectStoreInitialized(u)
 	if err != nil {
-		return errors.Wrap(err, "failed to detect if object store is initialized")
+		return errors.Wrapf(err, "failed to detect if object store %q is initialized", u.Spec.Store)
+	}
+
+	store, err := r.getObjectStore(u.Spec.Store)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get object store %q", u.Spec.Store)
 	}
 
+	objContext, err := object.NewMultisiteContext(r.context, r.clusterInfo, store)
+	if err != nil {
+		return errors.Wrapf(err, "Multisite failed to set on object context for object store user")
+	}
+	r.objContext = objContext
+	r.objContext.Endpoint = store.Status.Info["endpoint"]
```

Comment:
@leseb @travisn  I had to move around the assignment for this. Is it ok?

Reply:
Will `store.Status.Info["endpoint"]` always be set? If `Status` is not set, it could crash.

Reply:
It should be if the object-store is done reconciling here: https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/controller.go#L277. @leseb any comments?

Reply:
Why did you have to move it?

Reply:
I had to move it because `r.objContext` was not created yet in `getObjectStore()`. In one of the rounds of testing this assignment crashed an integration test.